### ISSUE TITLE
Option to disable Python2/comment-style type annotations

### ISSUE THREAD

**Issue by dfarley1** [python/mypy#13177]
Our codebase (all Python3) has a few places where we ended up writing comments that look like `# type: some description ...` prior to adding Mypy. They're usually within a larger multiline comment block when we're describing an expected Dict layout, e.g.
```
❯ cat foo.py
# Each Foo has the following fields:
# name: name of the foo
# display_name: human name
# type: type of foo it is
# tags: list of tags that apply to the foo
❯ mypy foo.py
foo.py:4: error: invalid syntax
Found 1 error in 1 file (errors prevented further checking)
```

Since we're only using Python3 and "proper" type annotations, it'd be nice if we could stop Mypy from trying to parse these types of comments. I know we could change our comment style to use dashes or something else that won't anger Mypy but it's annoying that it's necessary; and honestly colons are the better delimter anyways.

**Comment by hauntsaninja** [comment]
I'd welcome a PR adding an option for this!

Also see #12947 and the associated typing-sig discussion

**Comment by ronkorving** [comment]
Yes, please. I run into the same issue where completely innocent comments written for humans are being interpreted by `mypy` as a type annotation. I can imagine that turning off that feature completely could also make `mypy` run a bit faster?

**Comment by mbrancato** [comment]
Adding some sample code here since I ran across this in my issue. This not only applies to random comments, but also to commented out Python 3 code with PEP 526 type hints. I'm not sure why `mypy` processes these since there is no code prior to the comment.

```
# class A():
#     type: str
```

**Comment by spatinom** [comment]
can this be assigned to me?

**Comment by JelleZijlstra** [comment]
You are welcome to create a PR implementing this feature.

**Comment by jttimko5** [comment]
I will help work on this

**Comment by jttimko5** [comment]
Can someone please provide some guidance on where in the code base I should look to implement this new feature? I am new to the codebase so any help pointing me in the right direction would be super appreciated! Thanks!

**Comment by jttimko5** [comment]
Could i get some help locating where the syntax is checked for type annotations?

**Comment by ikonst** [comment]
Probably can be done by making these lines not run and have `converted = None` instead:
https://github.com/python/mypy/blob/70d4bb27a2733e43958dc5046befdc16ed297a5c/mypy/fastparse.py#L364-L366

**Comment by jttimko5** [comment]
> Probably can be done by making these lines not run and have `converted = None` instead:
> 
> https://github.com/python/mypy/blob/70d4bb27a2733e43958dc5046befdc16ed297a5c/mypy/fastparse.py#L364-L366

Could you elaborate on why these lines do?

**Comment by jttimko5** [comment]
Added pull request #15068 for this

**Comment by apoorva-01** [comment]
poked at this before doing a PR, one thing to sort out first.

the invalid syntax isn't mypy, it's CPython. mypy parses with `ast.parse(type_comments=True)` and CPython chokes on a stray `# type:` comment before mypy touches it. so the `converted = None` in translate_type_comment idea won't reach it, file's already dead.

`type_comments=False` fixes it, but CPython only fills `type_ignores` when it's True, and that's where mypy reads `# type: ignore`. so turning it off silently drops every `# type: ignore`. no good.

so the option's really "kill type comments as annotations, keep `# type: ignore`". two ways: parse with it off and re-scan `# type: ignore` ourselves (TYPE_IGNORE_PATTERN already exists), or keep it on and only retry-off on the stray-comment SyntaxError so real errors still show.

do you want py2-style `# type: int` gone too or just the stray case? and which approach? happy to build it once i know the shape. `--disable-type-comments` maybe
